File: gita/src-tauri/src/file_system.rs
// use std::fs::{self, File}; // Removed
// use std::io::{Read, Write}; // Removed
// use std::path::Path; // Removed
// use std::sync::Mutex; // Removed as it was likely for DB connection state or similar, not needed now

// Removed: use rusqlite::Connection;
// Removed: use tauri::AppHandle; // Was not present in snippet, but good to confirm
// Removed: use chrono::{DateTime, Utc};
// Removed: use regex::Regex; // Removed unused import
use serde::{Deserialize, Serialize};
// Removed: use uuid::Uuid;
// Removed: use walkdir::WalkDir;

#[derive(Debug, Serialize, Deserialize)]
struct NoteFrontMatter {
    id: Option<String>,
    title: Option<String>,
    created_at: Option<String>,
    updated_at: Option<String>,
    tags: Option<Vec<String>>,
}

impl Default for NoteFrontMatter {
    fn default() -> Self {
        NoteFrontMatter {
            id: None,
            title: None,
            created_at: None,
            updated_at: None,
            tags: None,
        }
    }
}

// All public functions (init_database, get_all_notes, search_notes, read_markdown_file,
// write_markdown_file, create_note, create_daily_note, delete_note, find_backlinks)
// and the public structs NoteMetadata and Note have been removed.
// The file now only contains NoteFrontMatter, its Default impl, and extract_front_matter.
// Necessary use statements (regex, serde, serde_yaml) are kept.
// Unnecessary use statements (fs, io, path, chrono, uuid, walkdir, rusqlite) are removed.

File: gita/src/api/fileSystem.ts
import { invoke } from '@tauri-apps/api/core';
import { Note, NoteMetadata } from '../types';

// Get the notes directory
export async function getNotesDirectory(): Promise<string> {
  return invoke('get_notes_directory');
}

// Set the notes directory
export async function setNotesDirectory(path: string): Promise<void> {
  return invoke('set_notes_directory', { path });
}

// Get the audio directory
export async function getAudioDirectory(): Promise<string> {
  return invoke('get_audio_directory');
}

// Set the audio directory
export async function setAudioDirectory(path: string): Promise<void> {
  return invoke('set_audio_directory', { path });
}

// Get all notes
export async function getAllNotes(): Promise<NoteMetadata[]> {
  return invoke('get_all_notes');
}

// Search notes
export async function searchNotes(query: string): Promise<NoteMetadata[]> {
  return invoke('search_notes', { query });
}

// Read a markdown file
export async function readMarkdownFile(path: string): Promise<Note> {
  return invoke('read_markdown_file', { path });
}

// Write a markdown file
export async function writeMarkdownFile(path: string, content: string): Promise<void> {
  return invoke('write_markdown_file', { path, content });
}

// Create a new note
export async function createNote(title: string, content: string): Promise<Note> {
  return invoke('create_note', { title, content });
}

// Create a daily note
export async function createDailyNote(): Promise<Note> {
  return invoke('create_daily_note');
}

// Delete a note
export async function deleteNote(noteId: string): Promise<void> {
  return invoke('delete_note', { note_id: noteId });
}

// Find backlinks for a note
export async function findBacklinks(noteId: string): Promise<NoteMetadata[]> {
  return invoke('find_backlinks', { note_id: noteId });
}

